### ISSUE TITLE
Pino: Allow logging trace metadata under different keys

### ISSUE THREAD

**Issue by drewcorlin1** [open-telemetry/opentelemetry-js-contrib#1664]
### Is your feature request related to a problem? Please describe

The current pino instrumentation from `@opentelemetry/instrumentation-pino` injects 3 fields into each log, `trace_id`, `span_id` and `trace_flags`. These fields are useful, but the snake case naming is slightly problematic in my case, where my company has strict logging standards and we'd like to use camelCase.

### Describe the solution you'd like to see
Allowing a different set of keys to be used in the config (this is assuming you are configuring the instrumentation via `getNodeAutoInstrumentations` from `@opentelemetry/auto-instrumentations-node`).

```typescript
'@opentelemetry/instrumentation-pino': {
  camelCase: true
}
```

### Describe alternatives you've considered
Reformatting the log manually, via the `logHook` config.

```typescript
'@opentelemetry/instrumentation-pino': {
    logHook: (span, record) => {
      // Reformat the injected log fields to use camelCase, eg. trace_id -> traceId
      const context = span.spanContext();
      record.traceId = context.traceId;
      record.spanId = context.spanId;
      record.traceFlags = context.traceFlags;

      // Be extra safe, in case someone manually logged these fields, even though they shouldn't
      if (record.trace_id === context.traceId) delete record.trace_id;
      if (record.span_id === context.spanId) delete record.span_id;
      if (record.trace_flags === context.traceFlags) delete record.trace_flags;
    },
  },
```

**Comment by Flarna** [comment]
I think it would be more flexible to allow to override the actual key names instead of adding just one more fixed set. e.g
```
'@opentelemetry/instrumentation-pino': {
  traceIdKey: 'traceId', // default is 'trace_id
  spanIdKey: 'span.id', // default is 'span_id'
  trace_flags: 'my-trace-flags'  // defaults to 'trace_flags'
}
```

**Comment by drewcorlin1** [comment]
Not stale

**Comment by mbrevda** [comment]
This could also be useful for specific log destinations. For example. GCP requires the following format: 

```js
   const trace = `projects/${projectId}/traces/${traceId}`;
      res['logging.googleapis.com/trace'] = trace;
    }
    res['logging.googleapis.com/trace_sampled'] = traceFlags;
    res['logging.googleapis.com/spanId'] = spanId;

**Comment by trentm** [comment]
@drewcorlin1 This can be closed now, right (it was handled by #1867)?

**Comment by drewcorlin1** [comment]
> @drewcorlin1 This can be closed now, right (it was handled by #1867)?

Yes!
